virttest/utils_misc.py

```diff
@@ -2795,51 +2843,86 @@ def format_windows_disk(session, did, mountpoint=None, size=None, fstype="ntfs")
+                                                  % kname)
+            size = size_output.splitlines()[0].split()[1]
+        all_disks_before = list_linux_guest_disks(session, True)
+        devname = line
+        logging.info("Create partition on disk '%s'" % devname)
+        mkpart_cmd = "parted -s %s mklabel gpt mkpart "
+        mkpart_cmd += "primary 0 %s"
+        mkpart_cmd = mkpart_cmd % (devname, size)
+        session.cmd_output_safe(mkpart_cmd)
+        session.cmd_output_safe("partprobe %s" % devname)
+        all_disks_after = list_linux_guest_disks(session, True)
+        partname = (all_disks_after - all_disks_before).pop()
+        logging.info("Format partition to '%s'" % fstype)
+        format_cmd = "yes|mkfs -t %s %s" % (fstype, partname)
+        session.cmd_output_safe(format_cmd)
+        if not mountpoint:
+            session.cmd_output_safe("mkdir /mnt/%s" % kname)
+            mountpoint = os.path.join("/mnt", kname)
+        logging.info("Mount the disk to '%s'" % mountpoint)
+        mount_cmd = "mount -t %s %s %s" % (fstype, partname, mountpoint)
+        session.cmd_output_safe(mount_cmd)
+        return True
 
     return False
 
 
-def format_guest_disk(session, did, mountpoint, size, fstype, ostype):
+def format_guest_disk(session, did, all_disks_did, ostype, partition=False,
+                      mountpoint=None, size=None, fstype=None):
     """
     Create a partition on disk in guest and format and mount it.
 
     :param session: session object to guest.
     :param did: disk ID in guest.
-    :param mountpoint: mount point for the disk.
-    :param fstype: filesystem type for the disk.
+    :param all_disks_did: a dict contains all disks did lists each
+                          include disk kname, serial and wwn for linux guest.
     :param ostype: guest os type 'windows' or 'linux'.
+    :param partition: if true, can format all disks; otherwise,
+                      only format the ones with no partition originally.
+    :param mountpoint: mount point for the disk.
+    :param size: partition size.
+    :param fstype: filesystem type for the disk; when it's the default None,
+                   it will use the default one for corresponding ostype guest
     :return Boolean: disk usable or not.
     """
-    if ostype == "windows":
-        return format_windows_disk(session, did, mountpoint, size, fstype)
-    return format_linux_disk(session, did, mountpoint, size, fstype)
+    if not fstype:
+        if ostype == "windows":
+            return format_windows_disk(session, did, mountpoint, size)
+        return format_linux_disk(session, did, all_disks_did, partition,
+                                 mountpoint, size)
+    else:
+        if ostype == "windows":
+            return format_windows_disk(session, did, mountpoint, size, fstype)
+        return format_linux_disk(session, did, all_disks_did, partition,
+                                 mountpoint, size, fstype)
```

Comment:
Since function format_guest_disk() is just a package of format_linux_disk() and format_windows_disk(), and the real default values are specified in the two detailed functions, so the if/else judgement is necessary; otherwise, the default value 'None' will be assigned.